src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/BackgroundDocumentGenerator.cs

```diff
@@ -233,15 +236,16 @@ private async Task Timer_TickAsync(CancellationToken cancellationToken)
             }
         }
 
-        private void NotifyDocumentsProcessed(KeyValuePair<string, DocumentSnapshot>[] work)
+        private void NotifyDocumentsProcessed(List<(RazorCodeDocument Output, DocumentSnapshot Document)> results)
         {
-            for (var i = 0; i < work.Length; i++)
+            for (var i = 0; i < results.Count; i++)
             {
                 foreach (var documentProcessedTrigger in _documentProcessedListeners)
                 {
-                    var document = work[i].Value;
+                    var document = results[i].Document;
+                    var codeDocument = results[i].Output;
 
-                    documentProcessedTrigger.DocumentProcessed(codeDocument: null, document);
```

Comment:
Out of curiosity was there ever a case beforehand where `DocumentProcessed(codeDocument...` was NOT null?

Reply:
Ya, it was hit or miss though. Basically it was dependent on whether or not GC ran inbetween the operations 😄